### PR TITLE
mds: add request to batch_op before taking auth pins and locks

### PR DESCRIPTION
**Pull request by ukernel**
MDS does not dispatch non-head requests in batch_op. So non-head
 requests in batch_op should not hold any authpins and locks. Otherwise
 head request may wait for these authpins/locks.
    
Fixes: https://tracker.ceph.com/issues/44785
Signed-off-by: "Yan, Zheng" <zyan@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/mds/Mutation.h [hunk 1]
```diff
@@ -435,12 +441,11 @@ struct MDRequestImpl : public MutationImpl {
   // indicates how may retries of request have been made
   int retry = 0;
 
-  bool is_batch_head = false;
+  std::map<int, std::unique_ptr<BatchOp> > *batch_op_map = nullptr;
```
Comment (batrick) [inline src/mds/Mutation.h:444]: Ownership is not clear to me. Should this be a `std::shared_ptr<std::map<...>>`?

Comment (ukernel) [reply src/mds/Mutation.h:444]: it's a pointer to either CInode::batch_ops or CDentry::batch_ops. can not be shared ptr

**Comment by ukernel** [comment]
re-pushed (only update commit message)

**Comment by batrick** [comment]
https://pulpito.ceph.com/?branch=wip-pdonnell-testing-20200711.001802

Failures are known/unrelated.

**Pull request merged**